### DISCUSSION TITLE
Think about JS library for generating graphs

### DISCUSSION THREAD

**Discussion by bschmalhofer** [RotherOSS/otobo#2664] [Ideas]
As far as I can see the JS libraries d3.js and nvd3 are used in OTOBO for generting graphs. The used versions seem to be from 2015. The development of nvd3.js seems to have stalled. d3.js is still in active development.
I suggest to find out whether d3.js is still state of the art. An extra bonus would be whether they can be used to generate images with node.js, so that OTOBO does not need a browser just for generating some images for the statistics reports.

**Comment by stefanhaerter** [comment]
If I recall it correctly, I used d3 on a previous job. This would be a great enhancement, totally support checking it.

**Comment by bschmalhofer** [comment]
See also #2300 .
